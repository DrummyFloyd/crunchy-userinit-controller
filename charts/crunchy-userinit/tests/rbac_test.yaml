# yaml-language-server: $schema=https://raw.githubusercontent.com/helm-unittest/helm-unittest/main/schema/helm-testsuite.json
suite: RBAC
templates:
  - rbac.yaml
release:
  name: test
  namespace: test-ns
chart:
  version: 0.0.0
  appVersion: 0.0.0
set:
  image:
    repository: ghcr.io/drummyfloyd/crunchy-userinit-controller
    tag: 0.0.0
tests:
  - it: should match snapshot with defautl
    asserts:
      - matchSnapshot: {}
