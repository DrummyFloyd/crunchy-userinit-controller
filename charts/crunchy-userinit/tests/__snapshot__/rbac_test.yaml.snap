should match snapshot with defautl:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRole
    metadata:
      name: test-crunchy-userinit-controller-role-cluster
    rules:
      - apiGroups:
          - kopf.dev
        resources:
          - clusterkopfpeerings
        verbs:
          - list
          - watch
          - patch
          - get
      - apiGroups:
          - apiextensions.k8s.io
        resources:
          - customresourcedefinitions
        verbs:
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - namespaces
        verbs:
          - list
          - watch
      - apiGroups:
          - admissionregistration.k8s.io/v1
          - admissionregistration.k8s.io/v1beta1
        resources:
          - validatingwebhookconfigurations
          - mutatingwebhookconfigurations
        verbs:
          - create
          - patch
      - apiGroups:
          - kopf.dev
        resources:
          - kopfexamples
        verbs:
          - list
          - watch
  2: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      name: test-crunchy-userinit-controller-role-namespaced
      namespace: test-ns
    rules:
      - apiGroups:
          - kopf.dev
        resources:
          - kopfpeerings
        verbs:
          - list
          - watch
          - patch
          - get
      - apiGroups:
          - ""
        resources:
          - events
        verbs:
          - create
      - apiGroups:
          - ""
        resources:
          - secrets
        verbs:
          - get
          - list
          - watch
          - update
          - patch
  3: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: ClusterRoleBinding
    metadata:
      name: test-crunchy-userinit-controller-rolebinding-cluster
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: ClusterRole
      name: test-crunchy-userinit-controller-role-cluster
    subjects:
      - kind: ServiceAccount
        name: test-crunchy-userinit-controller
        namespace: test-ns
  4: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: RoleBinding
    metadata:
      name: test-crunchy-userinit-controller-rolebinding-namespaced
      namespace: test-ns
    roleRef:
      apiGroup: rbac.authorization.k8s.io
      kind: Role
      name: test-crunchy-userinit-controller-role-namespaced
    subjects:
      - kind: ServiceAccount
        name: test-crunchy-userinit-controller
